src/actinia_core/resources/ephemeral_processing.py

```diff
@@ -1074,6 +1084,101 @@ def _run_module(self, process, poll_time=0.05):
+
+        Returns:
+            total: the size of the directory in bytes
+
+        """
+        total = 0
+        try:
+            for entry in os.scandir(directory):
+                if entry.is_file():
+                    total +=  os.path.getsize(entry)
+                elif entry.is_dir():
+                    total += self._get_directory_size(entry.path)
+        except NotADirectoryError:
+            return os.path.getsize(directory)
+        except PermissionError:
+            return 0
+        return total
+
+    def _save_interim_results(self):
+        """Saves the temporary mapset to the
+        `user_resource_interim_storage_path` by copying the directory or
+        rsyncing it
+        """
+        self.message_logger.info(
+            "Saving interim results of step %d" % self.progress_steps)
+        dest_base_path =  self.user_resource_interim_storage_path
+        dest = os.path.join(
+            dest_base_path, self.resource_id,
+            f"step{str(self.progress_steps)}")
+
+        if self.progress_steps == 1:
+            # copy temp mapset for first step
+            shutil.copytree(self.temp_mapset_path, dest)
+            self.message_logger.info(
+                "Maspset %s is copied" % self.temp_mapset_path)
+        else:
+            old_dest = os.path.join(
+                dest_base_path, self.resource_id,
+                f"step{str(self.progress_steps - 1)}")
+
+            # check if mapset has changed
+            cur_working_dir = os.getcwd()
+            cha512sum_cmd = "find . -type f -exec sha512sum {} \; | " + \
+                            "sort -k 2 | sha512sum"
+            # cha512sum of the previous step
+            os.chdir(old_dest)
+            p_cha512sum_prev = os.popen(cha512sum_cmd)
+            cha512sum_prev = p_cha512sum_prev.read().strip()
+            # cha512sum of the current step
+            os.chdir(self.temp_mapset_path)
+            p_cha512sum_curr = os.popen(cha512sum_cmd)
+            cha512sum_curr= p_cha512sum_curr.read().strip()
+            os.chdir(cur_working_dir)
+            del cur_working_dir
+            # compare cha512sums
+            if cha512sum_prev == cha512sum_curr:
+                self.message_logger.info(
+                    "Cha512sum of maspsets are equal; renaming interim result")
+                os.rename(old_dest, dest)
+                return
+
+            # get folder sizes in bytes
+            size_prev_step = self._get_directory_size(old_dest)
+            size_curr_step = self._get_directory_size(self.temp_mapset_path)
+
+            # rsync or copy folder
+            if size_curr_step > size_prev_step * 0.9:
+                self.message_logger.info("Copy old interim result")
+                shutil.copytree(old_dest, dest)
+            self.message_logger.info(
+                "Rsync mapset %s to interim result" % self.temp_mapset_path)
+            rsync_cmd = [
+                "rsync",
+                "-rzP",
```

Comment:
```suggestion
                "--recursive --compress --partial --progress --times",
```

Added `--times` to preserve file modification times (time stamps)
AFAIK the old single dash flags will go away one day and the double dash flags are preferred (and more readable).